src/contracts/socket/index.ts

```diff
@@ -0,0 +1,166 @@
+    userAddress: cowShedAccount, // bridge input token will be in cowshed account
+    recipient: recipient,
+    sort: 'output', // optimize for output amount
+    singleTxOnly: true, // should be only single txn on src chain, no destination chain txn
+    isContractCall: true, // get quotes that are compatible with contracts
+    disableSwapping: true, // should not show routes that require swapping
+    includeBridges,
+  });
+  if (!quote) {
+    throw new Error('No quote found');
+  }
+  console.log('🔗 Socket quote:', quote);
+  // check if routes are found
+  if (!quote.result.routes.length) {
+    throw new Error('No routes found');
+  }
+  // check if only single user tx is present
+  if (quote.result.routes[0].userTxs.length > 1) {
+    throw new Error('Multiple user txs found');
+  }
+  // check if the user tx is fund-movr
+  if (quote.result.routes[0].userTxs[0].userTxType !== 'fund-movr') {
+    throw new Error('User tx is not fund-movr');
+  }
+
+  // use the first route to prepare the bridge tx
+  const route = quote.result.routes[0];
+  const txData = await getBungeeRouteTransactionData(route);
+  const { routeId, encodedFunctionData } = decodeBungeeTxData(
+    txData.result.txData
+  );
+
+  // Create bridged token contract
+  const bridgedTokenContract = WeirollContract.createContract(
+    getErc20Contract(sourceToken),
+    CommandFlags.CALL
+  );
+
+  // Get balance of CoW shed proxy
+  console.log(
+    `[socket] Get cow-shed balance for ERC20.balanceOf(${cowShedAccount}) for ${bridgedTokenContract.address}`
+  );
+
+  // Get bridged amount (balance of the intermediate token at swap time)
+  const sourceAmountIncludingSurplusBytes = planner.add(
+    bridgedTokenContract.balanceOf(cowShedAccount).rawValue()
+  );
+
+  // Check & set allowance for SocketGateway to transfer bridged tokens
+  // check if allowance is sufficient
+  const {
+    approvalData: {
+      approvalTokenAddress,
+      allowanceTarget,
+      minimumApprovalAmount,
+    },
+  } = txData.result;
+  const intermediateTokenContract = getErc20Contract(
+    approvalTokenAddress,
+    await getWallet(sourceChain)
+  );
+  const allowance = await intermediateTokenContract.allowance(
+    cowShedAccount,
+    allowanceTarget
+  );
+  console.log('current cowshed allowance', allowance);
+  if (allowance < minimumApprovalAmount) {
+    // set allowance
+    const approvalTokenContract = WeirollContract.createContract(
+      getErc20Contract(approvalTokenAddress),
+      CommandFlags.CALL
+    );
+    console.log(
+      `[socket] approvalTokenContract.approve(${allowanceTarget}, ${sourceAmountIncludingSurplusBytes}) for ${approvalTokenContract}`
+    );
+    const allowanceToSet = ethers.utils.parseUnits(
+      '1000',
+      await intermediateTokenContract.decimals()
+    );
+    planner.add(approvalTokenContract.approve(allowanceTarget, allowanceToSet));
+  }
+
+  const bytesLibContractAddress = '0x8f6BA63528De7266d8cDfDdec7ACFA8446c62aB4';
```

Comment:
Will this contract be the same in every network?

Reply:
I don't see anuything in Ethereum for example https://etherscan.io/address/0x8f6BA63528De7266d8cDfDdec7ACFA8446c62aB4

Reply:
This is currently only deployed on Arbiturm for this PoC. Need to deploy on all chains and will share a mapping or deploy at a deterministic address everywhere